pkg/apis/core/validation/validation_patch.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+var (
+	// we make an exception for the following secret objects, during update
+	// we allow the secret type to mutate from:
+	//     "SecretTypeTLS" -> "kubernetes.io/tls"
+	whitelist = map[string]struct{}{
+		"openshift-kube-apiserver-operator/aggregator-client-signer":          {},
+		"openshift-kube-apiserver-operator/kube-apiserver-to-kubelet-signer":  {},
+		"openshift-kube-apiserver-operator/localhost-serving-signer":          {},
+		"openshift-kube-apiserver-operator/service-network-serving-signer":    {},
+		"openshift-kube-apiserver-operator/loadbalancer-serving-signer":       {},
+		"openshift-kube-apiserver-operator/localhost-recovery-serving-signer": {},
+		"openshift-kube-apiserver-operator/kube-control-plane-signer":         {},
+		"openshift-kube-apiserver-operator/node-system-admin-signer":          {},
+	}
```

Comment:
- https://github.com/openshift/cluster-etcd-operator/blob/eeef8033eb058b0f9899c7b16aead237d5b84462/manifests/0000_20_etcd-operator_03_secret.yaml#L9-L10
- https://github.com/openshift/cluster-kube-controller-manager-operator/blob/bdb42fd60a64adf6f98cc7be4d90a76ea30662d6/bindata/bootkube/manifests/secret-csr-signer-signer.yaml#L4-L5

Reply:
Are these secrets also affected ? Or do we want to transition them to the new type without deletion?